app/src/main/java/ch/epfl/sweng/eventmanager/repository/UserRepository.java

```diff
@@ -0,0 +1,55 @@
+package ch.epfl.sweng.eventmanager.repository;
+
+import android.os.Build;
+import android.support.annotation.RequiresApi;
+
+import ch.epfl.sweng.eventmanager.repository.data.User;
+
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+// FIXME: check if we can do something intelligent with dependency injection?
+@Singleton
+public class UserRepository {
+    private final List<User> USERS = new ArrayList<>(1);
+    private static UserRepository instance = null;
+
+    {
+        USERS.add(
+               new User(1, "Lamb Da", "lamb.da@domain.tld")
+        );
+    }
+
+   private UserRepository() {
+      // Exists only to defeat instantiation.
+   }
+
+   public static UserRepository getInstance() {
+      if(instance == null) {
+         instance = new UserRepository();
+      }
+      return instance;
+   }
+
+    public List<User> getUsers() {
+        return USERS;
+    }
+
+    // FIXME: rewrite it to be compatible with older APIs revisions
```

Comment:
What in this method isn't compatible with older APIs?

Reply:
It looks like streams and `Optional.ofNullable/1` require API 24 (current minimum is 15, see #42). I didn't dig in the documentation for details, this PR is still WIP.